src/Tokenizer/Tokens.php

```diff
@@ -859,20 +860,28 @@ public function insertSlices(array $slices): void
 
         krsort($slices);
 
+        if (array_key_first($slices) > $oldSize) {
+            throw new \OutOfBoundsException('Cannot insert outside of collection.');
```

Comment:
I'm not convinced it's good validation, as it's checking only index for first slice. And it looks like there are not tests checking it? :(

ref #6172

Reply:
we sort the keys above it, so we know all other keys are of a lower value so we only need to test the first,
all others are tested to be greater or equal to zero